### PR TITLE
Task 5 (AWS Simple Storage Service - S3)

### PR DESCRIPTION
**Pull request by shurygindv**
- [x]  **done** Task 5.1 (lambda importProductsFile, signedUrl and etc.)
- [x]   **done** Task 5.2 (create importFileParser, streams and etc.)  
- [x]  **done** Additional requirements (FE integration, async / tests / `parsed`  folder after transformations)

`bucket` will be created automatically according to `serverless.yml` resources, also, you may update the frontend page in few seconds after product imports and check results (for demo)

**Frontend** 
Product import (in panel) : https://d7cnyur3qbvu2.cloudfront.net/admin/products
MR: https://github.com/shurygindv/nodejs-aws-fe/pull/4

**Backend**
 `GET` Signed URL: https://jkzgfse086.execute-api.eu-west-1.amazonaws.com/dev/import (required `name` query)

### CODE REVIEW

File: import-service/serverless.yml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+resources:
+  Resources:
+    # -> https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-s3-bucket.html
+    CSVBucket:
+      Type: AWS::S3::Bucket
+      Properties:
+        AccelerateConfiguration: 
+          AccelerationStatus: Suspended
+        BucketName: ${self:provider.environment.BUCKET_NAME}
+        CorsConfiguration:
```
Comment (viacheslav-karnaukh) [inline import-service/serverless.yml:77]: good to see it here

File: import-service/src/libs/validator.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+const PNG_OR_JPG_REGEX = /^[a-zA-Z0-9]{3,}\.(jpg|png)$/i;
+const CSV_FILE_NAME_REGEX = /^[a-zA-Z0-9]+\.csv$/i;
+const GUID_REGEX = /^[0-9A-Fa-f\-]{36}$/;
+
+const isImageFileName = (n: string) => PNG_OR_JPG_REGEX.test(n);
+const hasProperty = <T>(obj: T, name: string) => obj.hasOwnProperty(name);
+const isGuid = (v: string) => GUID_REGEX.test(v);
+const isCSVFileName = (n: string) => CSV_FILE_NAME_REGEX.test(n);
```
Comment (viacheslav-karnaukh) [inline import-service/src/libs/validator.ts:8]: some csv names cannot be passed by this regex. e.g. I had a couple of such files, they were not passed. `1,2,3.csv` and `test - Copy.csv`

**Review by viacheslav-karnaukh — APPROVED** [review]

File: import-service/src/libs/validator.ts (hunk 2 above)
Comment (shurygindv) [reply import-service/src/libs/validator.ts:8]: an eagle eye)) thank you) fixed/deployed, specified variants work as well

File: import-service/serverless.yml (hunk 1 above)
Comment (shurygindv) [reply import-service/serverless.yml:77]: )))) it let me create a bucket without any headache with access and avoid manual setup in future

File: import-service/src/libs/validator.ts (hunk 2 above)
Comment (shurygindv) [reply import-service/src/libs/validator.ts:8]: you could try to repeat your upload, it works

**Pull request merged**